### PR TITLE
[chisel] Add support for Chisel Binder framework

### PR DESCRIPTION
**Pull request by SpriteOvO**

### CODE REVIEW

**Comment by SpriteOvO** [comment]
Run command `nix build .#t1.v1024-l8-b2.elaborate-binder` to check the results produced by Chisel Binder framework.

**Comment by SpriteOvO** [comment]
CC @sequencer for review, thanks.

File: build.sc [hunk 1]
```diff
@@ -9,6 +9,7 @@ import mill.scalalib.scalafmt._
 import mill.scalalib.TestModule.Utest
 import coursier.maven.MavenRepository
 import $file.dependencies.chisel.build
+import $file.dependencies.chisel.common
```
Comment (sequencer) [inline build.sc:12]: Don't introduce common.

File: build.sc [hunk 2]
```diff
@@ -248,11 +249,22 @@ trait FPGA
...
+
+  def scalaVersion = T(v.scala)
+}
+
 // Module to generate RTL from json config
 object elaborator extends Elaborator
 
 trait Elaborator
-  extends millbuild.common.ElaboratorModule {
+  extends millbuild.common.ElaboratorModule with millbuild.dependencies.chisel.common.HasCIRCTPanamaBinderModule {
```
Comment (sequencer) [inline build.sc:267]: vendor a `HasCIRCTPanamaBinderModule` in our own `common.sc`

File: elaborator/src/Main.scala [hunk 3]
```diff
@@ -13,12 +15,13 @@ object Main {
 
   @main
   case class ElaborateConfig(
-    @arg(name = "target-dir", short = 't') targetDir: os.Path) {
+    @arg(name = "target-dir", short = 't') targetDir: os.Path,
+    @arg(name = "binder") binder: Boolean) {
```
Comment (sequencer) [inline elaborator/src/Main.scala:19]:
```suggestion
    @arg(name = "binder", short = 'b') binder: Boolean) {
```

File: elaborator/src/Main.scala [hunk 4]
```diff
@@ -29,6 +32,9 @@ object Main {
             topName = circuit.main
             os.write(targetDir / s"$topName.fir", circuit.serialize)
             None
+          case chisel3.internal.panama.circt.PanamaCIRCTConverterAnnotation(converter) =>
+            converter.exportSplitVerilog(dir_)
```
Comment (sequencer) [inline elaborator/src/Main.scala:36]: ?

**Review by sequencer — CHANGES_REQUESTED** [review]

**Pull request closed without merging**